src/Symfony/Component/Notifier/Bridge/OneSignal/Tests/OneSignalTransportTest.php

```diff
@@ -146,4 +146,32 @@ public function testSend()
 
         $this->assertSame('b98881cc-1e94-4366-bbd9-db8f3429292b', $sentMessage->getMessageId());
     }
+
+    public function testSendExternalIds()
+    {
+        $response = $this->createMock(ResponseInterface::class);
+        $response->expects($this->exactly(2))
+            ->method('getStatusCode')
+            ->willReturn(200);
+        $response->expects($this->once())
+            ->method('getContent')
+            ->willReturn(json_encode(['id' => 'b98881cc-1e94-4366-bbd9-db8f3429292b', 'recipients' => 1, 'external_id' => null]));
+
+        $expectedBody = json_encode(['app_id' => '9fb175f0-0b32-4e99-ae97-bd228b9eb246', 'headings' => ['en' => 'Hello'], 'contents' => ['en' => 'World'], 'include_aliases' => ['external_id' => ['ea345989-d273-4f21-a33b-0c006efc5edb']], 'target_channel' => 'push']);
+
+        $client = new MockHttpClient(function (string $method, string $url, array $options = []) use ($response, $expectedBody): ResponseInterface {
+            $this->assertJsonStringEqualsJsonString($expectedBody, $options['body']);
+
+            return $response;
```

Comment:
I think it's better to use JsonMockResponse from Symfony HttpClient

Reply:
As return in the MockClient? Not familiar with it, probably passed by me.

Tbh I only copied and adapted the existing test, I can update them though

Reply:
I think @OskarStark streamlined tests to use it, not sure why it's not used there :)

Reply:
yeah figured it out and changed the tests :)

Reply:
Though... It's strange that the tests not fail since I forgot to change the HTTP status on the error ones

Reply:
Found why, the Transport throws the same exception when ID is not set, so that's why it didn't notice.
Can't figure out a good way to check for it though without changing the exception text